warn-stranded.lua

```diff
@@ -0,0 +1,581 @@
+        if unitIgnored(unit, ignored) then
+            print('Unit '..unit.id..' already ignored, doing nothing to them.')
+        else
+            print('Ignoring unit '..unit.id)
+            toggleUnitIgnore(unit, ignored)
+        end
+    end
+
+    return true
+end
+
+local function unignoreGroup(groups, groupNumber)
+    local ignored = deserializeIgnoredUnits()
+
+    if groupNumber > #groups then
+        print('Group '..groupNumber..' does not exist')
+        return false
+    end
+
+    if groups[groupNumber]['mainGroup'] then
+        print('Group '..groupNumber..' is the main group of dwarves. Unignoring.')
+    end
+
+    for _, unit in ipairs(groups[groupNumber]['units']) do
+        if unitIgnored(unit, ignored) then
+            print('Unignoring unit '..unit.id)
+            ignored = toggleUnitIgnore(unit, ignored)
+        else
+            print('Unit '..unit.id..' not already ignored, doing nothing to them.')
+        end
+    end
+
+    return true
+end
+
+function doCheck()
+    local result, strandedGroups = getStrandedUnits()
+
+    if result then
+        return warning{groups=strandedGroups}:show()
+    end
+end
+
+if dfhack_flags.module then
+    return
+end
+
+if not dfhack.isMapLoaded() then
+    qerror('warn-stranded requires a map to be loaded')
+end
+
+-- =========================================================================
+--                       Command Line Interface
+-- =========================================================================
+
+local positionals = argparse.processArgsGetopt(args, {})
+
+if positionals[1] == 'clear' then
+    print('Clearing unit ignore list.')
+    return clear()
+end
+
+local parameter = tonumber(positionals[2])
+
+if positionals[1] == 'status' then
+    local result, strandedGroups = getStrandedUnits()
+
+    if result then
+        local ignoresCache = deserializeIgnoredUnits()
+
+        for groupIndex, group in ipairs(strandedGroups) do
+            local groupDesignation = getGroupDesignation(group, groupIndex, true)
+
+            for _, unit in ipairs(group['units']) do
+                local text = ''
+
+                text = addIgnored(text, unit, ignoresCache)
+                text = addId(text, unit)
+
+                print(text..getUnitDescription(unit)..groupDesignation)
+            end
+        end
+
+        return true
+    end
+
+
+    print('No citizens are currently stranded.')
+
+    -- We have some ignored citizens
+    if not (next(strandedGroups) == nil) then
+        print('\nIgnored citizens:')
+
+        for walkGroup, units in pairs(strandedGroups) do
+            for _, unit in ipairs(units) do
+                local text = ''
+
+                text = addId(text, unit)
+                text = text..getUnitDescription(unit)..' {'..walkGroup..'}'
+
+                print(text)
+            end
+        end
+    end
+
+    return false
+end
+
+if positionals[1] == 'ignore' then
```

Comment:
a more standard idiom here would be to use `elseif` instead of `end if`, especially since these cases are mutually exclusive